### PR TITLE
add mongoose and connect to mongodb

### PR DESCRIPTION
**Pull request by xddenise**
- [x] All features are tested
- [x] Version is bumped
- [x] Changelog is updated
- [x] Documentation is updated if needed

### CODE REVIEW

**Comment by xddenise** [comment]
Can use postman to test the /user endpoint with POST with this json, which should add an instance into your local database inside: database "project", collection "users":
```
{
    "username": "AAA",
    "password": "BBB",
    "gender": "female",
    "major": "computer_science",
    "year": "3",
    "tags": [
        "BBB"
    ],
    "savedPostIds": [],
    "hiddenPostIds": []
}
```

File: src/App.ts [hunk 1]
```diff
@@ -1,12 +1,42 @@
...
+import {User} from "./backend/Models";
+import bodyParser from 'body-parser';
+const mongoose = require('mongoose');
 
 const app = express();
 const port = 3000;
+app.use(bodyParser.json());
+
+// mongoose connection, connect to local database with db name "project", creates new database if "project" doesn't exist
+mongoose.connect('mongodb://localhost:27017/project', {useNewUrlParser: true});
```
Comment (ZijiaZhang) [inline src/App.ts:12]: Maybe we can make this configureable so we can use a separate database in tests?

File: src/App.ts [hunk 2]
```diff
@@ -1,12 +1,42 @@
...
+});
 
 app.use(express.static(path.join(__dirname, '..', 'public')));
+
+// TODO: will move to a different file after rebasing with backend apiRouter setup PR
+// Sample apis for User Schema:
+app.post('/user', (req, res) => {
+    User.create(req.body, (err: Error, user: any) => {
+        console.log(err, user);
+        res.send(user);
```
Comment (ZijiaZhang) [inline src/App.ts:26]: Also need handle error here?

Comment (xddenise) [reply src/App.ts:26]: How are we planning to deal with errors? Log and send back to frontend to display for user?

File: src/App.ts (hunk 1 above)
Comment (xddenise) [reply src/App.ts:12]: Yep sure

**Pull request merged**